Renaming Method issues an error

Hi, asking the server to rename **StaticBLock** to StaticB**l**ock raises an error:

eg: 
``` Java
class SDemo3 {
    public static void main(String[] args) {
	StaticBLock ob = new StaticBLock("Inside Constructor");

	System.out.println("Square root of 2 is " +
			   StaticBLock.rootOf2);
	System.out.println("Square root of 3 is " +
			   StaticBLock.rootOf3);
    }
}
```

```
jsonrpc-error: "request id=11 failed:", (jsonrpc-error-code . -32603), (jsonrpc-error-message . "Internal error."), (jsonrpc-error-data . "java.util.concurrent.CompletionException: java.lang.NullPointerException
	at java.base/java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:314)
	at java.base/java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:319)
	at java.base/java.util.concurrent.CompletableFuture$UniApply.tryFire(CompletableFuture.java:645)
	at java.base/java.util.concurrent.CompletableFuture$Completion.exec(CompletableFuture.java:479)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1020)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1656)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1594)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:177)
Caused by: java.lang.NullPointerException
	at org.eclipse.jdt.ls.core.internal.preferences.ClientPreferences.isResourceOperationSupported(ClientPreferences.java:185)
	at org.eclipse.jdt.ls.core.internal.ChangeUtil.convertTextEdit(ChangeUtil.java:220)
	at org.eclipse.jdt.ls.core.internal.ChangeUtil.convertTextChange(ChangeUtil.java:209)
	at org.eclipse.jdt.ls.core.internal.ChangeUtil.doConvertCompositeChange(ChangeUtil.java:132)
	at org.eclipse.jdt.ls.core.internal.ChangeUtil.convertCompositeChange(ChangeUtil.java:87)
	at org.eclipse.jdt.ls.core.internal.handlers.RenameHandler.rename(RenameHandler.java:95)
	at org.eclipse.jdt.ls.core.internal.handlers.JDTLanguageServer.lambda$21(JDTLanguageServer.java:666)
	at org.eclipse.jdt.ls.core.internal.handlers.JDTLanguageServer.lambda$26(JDTLanguageServer.java:793)
	at java.base/java.util.concurrent.CompletableFuture$UniApply.tryFire(CompletableFuture.java:642)
	... 6 more
")`
```
https://github.com/joaotavora/eglot/issues/179#issuecomment-445499360

Its seems that: "If we don't provide **resourceOperations** and **failureHandling** keys in initialization options, the server errors out with"


Eclipser Server version: org.eclipse.equinox.launcher.gtk.linux.x86_64_1.1.900.v20180922-1751.jar
Debian SID
Emacs 26.1 / Eglot

--- a/org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/preferences/ClientPreferences.java
+++ b/org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/preferences/ClientPreferences.java
@@ -181,7 +181,9 @@ public class ClientPreferences {
 
 	public boolean isResourceOperationSupported() {
 		//@formatter:off
-		return capabilities.getWorkspace() != null && capabilities.getWorkspace().getWorkspaceEdit() != null
+		return capabilities.getWorkspace() != null
+				&& capabilities.getWorkspace().getWorkspaceEdit() != null
+				&& capabilities.getWorkspace().getWorkspaceEdit().getResourceOperations() != null
 				&& capabilities.getWorkspace().getWorkspaceEdit().getResourceOperations().contains(ResourceOperationKind.Create)
 				&& capabilities.getWorkspace().getWorkspaceEdit().getResourceOperations().contains(ResourceOperationKind.Rename)
 				&& capabilities.getWorkspace().getWorkspaceEdit().getResourceOperations().contains(ResourceOperationKind.Delete);

--- a/org.eclipse.jdt.ls.tests/src/org/eclipse/jdt/ls/core/internal/handlers/InitHandlerTest.java
+++ b/org.eclipse.jdt.ls.tests/src/org/eclipse/jdt/ls/core/internal/handlers/InitHandlerTest.java
@@ -66,6 +66,7 @@ import org.eclipse.lsp4j.TextDocumentClientCapabilities;
 import org.eclipse.lsp4j.TextDocumentSyncKind;
 import org.eclipse.lsp4j.TextDocumentSyncOptions;
 import org.eclipse.lsp4j.WorkspaceClientCapabilities;
+import org.eclipse.lsp4j.WorkspaceEditCapabilities;
 import org.eclipse.lsp4j.jsonrpc.messages.Either;
 import org.junit.After;
 import org.junit.Before;
@@ -264,6 +265,16 @@ public class InitHandlerTest extends AbstractProjectsManagerBasedTest {
 			FileUtils.deleteDirectory(targetFile);
 			FileUtils.deleteDirectory(tempDirectory);
 		}
+	}
+
+	@Test
+	public void testMissingResourceOperations() throws Exception {
+		ClientCapabilities capabilities = new ClientCapabilities();
+		WorkspaceClientCapabilities worspaceCapabilities = new WorkspaceClientCapabilities();
+		worspaceCapabilities.setWorkspaceEdit(new WorkspaceEditCapabilities());
+		capabilities.setWorkspace(worspaceCapabilities);
+		ClientPreferences preferences = new ClientPreferences(capabilities);
+		assertFalse(preferences.isResourceOperationSupported());
 	}
 
 	private void removeExclusionPattern(IJavaProject javaProject) throws JavaModelException {

